client/state/signup/progress/reducer.js

```diff
@@ -51,8 +53,23 @@ function processStep( state, { step } ) {
 }
 
 function removeUnneededSteps( state, { flowName } ) {
-	const flowSteps = get( flows, `${ flowName }.steps`, [] );
-	return state.filter( step => flowSteps.includes( step.stepName ) );
+	let flowSteps = [];
+	const user = userFactory();
+
+	flowName =
+		'onboarding' === abtest( 'improvedOnboarding' ) && 'ecommerce' === flowName
+			? flowName + '-onboarding'
+			: flowName;
```

Comment:
That's one long ternary expression! Can we make it something like `if(x) { flowName = 'ecommerce-onboarding'; }`?

Reply:
Thanks, fixed in [8ce58b6](https://github.com/Automattic/wp-calypso/pull/28672/commits/8ce58b6eea60717f8758c44edef851aaa3e13e38.)